fdbclient/BlobCipher.cpp

```diff
@@ -1911,7 +1911,9 @@ void testKeyCacheEssentials(DomainKeyMap& domainKeyMap,
 		uint8_t rawCipher[baseCipher->len];
 		memcpy(rawCipher, baseCipher->key.get(), baseCipher->len);
 		// modify few bytes in the cipherKey
-		for (int i = 2; i < 5; i++) {
+		int startIdx = deterministicRandom()->randomInt(0, 2);
+		int endIdx = deterministicRandom()->randomInt(2, std::min(baseCipher->len, 5));
```

Comment:
We should add 1 to the end bounds of the random int, because it is exclusive and then the resulting value is exclusive in the loop.

```suggestion
		int endIdx = deterministicRandom()->randomInt(2, std::min(baseCipher->len, 5) + 1);
```

Reply:
I think idea is to flip few bits, we need not to flip all the bits. The current fix should work fine.

Reply:
It changes the behavior (It will now go up through 4 instead of 5), but that's ok if you are happy with it. However, if you do this you will need to assert that `baseCipher->len >=3` rather than 2 as proposed above.

Reply:
The min length of the baseCipher in the test is 4 and it is random number between (4, 128]
`len(deterministicRandom()->randomInt(4, 128))`

Also, I've some more changes to fix some unrelated issues to this PR, but issues due to cipher buffer greater than underlying AES supported library. I've removed the hardcoded values with `named consts`. For immediate fix, this should work, however, will add an assert to ensure `size >= 3`

Reply:
What I mean is that whereas before this could change up through byte[4] (i.e. the 5th byte) if the length was long enough, it will now only change up through byte[3] (i.e. the 4th byte). That is because the maximum int you can generate for `endIdx` is 4. 

I agree that this works, but it also seems a little bit weird because you won't actually change the last byte of a 4-byte cipher. That doesn't seem wrong, but it does seem unintentional in this implementation and it seems like what was actually intended was what I wrote above.

Reply:
Initial loop was change 3 bytes
`for (int i = 2; i < 5; i++) {` 

However, earlier the cipher used to be 32 bytes fix length buffer, which I recently updated to be random length buffer (between 4-128 bytes). I'm happy to make the changes as suggested, but even flipping 1 bit is good enough is what I'm saying :)

Reply:
I think you can keep it at `2` unless you make AJs suggestion

Reply:
Actually it is better to simply the test and just flip one bit as it is more stronger test in any case than multiple bit flip.

Reply:
I understand, but I think the concern here is that what the code is doing unintentional, and that makes it harder to understand. The most straightforward change to the existing code is to keep iterating < 5 unless 5 is too large, and that's what my proposal does. The new proposal appears to be avoiding the last byte of the cipher and also decreases how far it iterates but without explanation. Both cases still work, but the former seems like a more straightforward change.